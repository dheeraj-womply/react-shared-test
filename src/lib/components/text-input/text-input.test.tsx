import React from "react"
import renderer from "react-test-renderer"
import { TextInput } from "./index"
import { InputAdornment } from '@material-ui/core'

test('Text Input elements are OK', () => {
    const component = renderer.create(
        <TextInput
            label={'Label text'}
            variant="outlined"
            value={'Input value'}
            fullWidth
        />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})

test('Text Password Input elements are OK', () => {
    const component = renderer.create(
        <TextInput
            label={'Current Password'}
            variant="outlined"
            isPassword={true}
            value={'Input value'}
            fullWidth
        />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})

test('Text Input with error and helper text are OK', () => {
    const component = renderer.create(
      <TextInput
        label={"Current Password"}
        variant="outlined"
        value={"Input value"}
        fullWidth
        inputProps={{
          "data-testid": "text-input",
        }}
        InputProps={{
          startAdornment: <InputAdornment position="start">+1</InputAdornment>,
        }}
        helperText={"Required"}
        isPassword={true}
        error={true}
      />
    );
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
