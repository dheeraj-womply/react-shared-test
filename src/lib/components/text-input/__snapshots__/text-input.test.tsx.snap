// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Input elements are OK 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
    data-shrink={true}
  >
    Label text
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
    onClick={[Function]}
    style={
      Object {
        "MozAppearance": "none",
        "WebkitAppearance": "none",
        "appearance": "none",
        "boxShadow": "none",
      }
    }
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value="Input value"
    />
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
      >
        <span>
          Label text
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Text Input with error and helper text are OK 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-error Mui-error MuiFormLabel-filled"
    data-shrink={true}
  >
    Current Password
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
    onClick={[Function]}
    style={
      Object {
        "MozAppearance": "none",
        "WebkitAppearance": "none",
        "appearance": "none",
        "boxShadow": "none",
      }
    }
  >
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionStart"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        +1
      </p>
    </div>
    <input
      aria-invalid={true}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
      data-testid="text-input"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="password"
      value="Input value"
    />
    <button
      aria-label="toggle password visibility"
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="-1"
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
          />
        </svg>
      </span>
    </button>
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
      >
        <span>
          Current Password
        </span>
      </legend>
    </fieldset>
  </div>
  <p
    className="MuiFormHelperText-root MuiFormHelperText-contained Mui-error MuiFormHelperText-filled"
  >
    Required
  </p>
</div>
`;

exports[`Text Password Input elements are OK 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
    data-shrink={false}
  >
    Current Password
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
    onClick={[Function]}
    style={
      Object {
        "MozAppearance": "none",
        "WebkitAppearance": "none",
        "appearance": "none",
        "boxShadow": "none",
      }
    }
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="password"
      value="Input value"
    />
    <button
      aria-label="toggle password visibility"
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="-1"
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
          />
        </svg>
      </span>
    </button>
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="PrivateNotchedOutline-legendLabelled-3"
      >
        <span>
          Current Password
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
